src/im/imstaff.cpp

```
@@ -128,10 +124,10 @@ void CalcWinFeatures(const imImage* image, float *features, int position_v, int
 	features[0] = 1.0 / (region_count + 1); // global
 
 	// centroid
-	float *cx = (float*)malloc(region_count*sizeof(float));
-	memset( cx, 0, region_count*sizeof(float));
-	float *cy = (float*)malloc(region_count*sizeof(float));
-	memset( cy, 0, region_count*sizeof(float));
+	double *cx = (double*)malloc(region_count*sizeof(double));
+	memset( cx, 0, region_count*sizeof(double));
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)